Console/Command/CakePHPUtilShell.php

```diff
@@ -0,0 +1,330 @@
+		} elseif (!empty($this->_exceptions)) {
+			// Expected shutdown with Exceptions
+			$this->out('The ' . $trace[1]['class'] . ' has terminated the \'' . $component . '/' . $action . '\' action with exit code ' . $code . '. The error occurred on line ' . $trace[0]['line'] . '. Additionally, the following Exceptions were thrown: ' . "\n\n" . implode("\r\n", $this->_exceptions));
+		} else {
+			$this->out(sprintf('The %s has ran successfully. No exceptions were thrown.', $trace[1]['class']));
+		}
+		$this->hr();
+
+		// Exit the Shell
+		parent::_stop($code);
+	}
+
+/**
+ * Print warnings that were catched by the AppShellErrorHandler.
+ */
+	protected function _printWarnings() {
+		if (AppShellErrorHandler::hasWarnings()) {
+			$warnings = AppShellErrorHandler::flushWarnings();
+			$this->out("The following warnings occurred during the cronjob:\n");
+			foreach ($warnings as $warning) {
+				$this->out(trim($warning[1]));
+				foreach ($warning[2] as $backtrace) {
+					// The warning_handler doesn't have a a file and line number in the backtrace.
+					if (!isset($backtrace['line']) && !isset($backtrace['file'])) {
+						continue;
+					}
+					$this->out($backtrace['class'] . $backtrace['type'] . $backtrace['function'] . ' in ' . $backtrace['file'] . ' on line ' . $backtrace['line']);
+				}
+				$this->out("\n");
+			}
+			if (AppShellErrorHandler::isListFull()) {
+				$this->out('The messages buffer is full, no further messaged were cached.');
+			}
+		} else if(empty($this->_exceptions)) {
+			$this->out(self::SUCCESS_MESSAGE);
+		}
+	}
+
+	/**
+	 * Does the exact same thing as parent::out(), but it checks if the $this->debug is set to 1
+	 * Outputs a single or multiple messages to stdout. If no parameters
+	 * are passed outputs just a newline.
+	 *
+	 * ### Output levels
+	 *
+	 * There are 3 built-in output level.  Shell::QUIET, Shell::NORMAL, Shell::VERBOSE.
+	 * The verbose and quiet output levels, map to the `verbose` and `quiet` output switches
+	 * present in  most shells.  Using Shell::QUIET for a message means it will always display.
+	 * While using Shell::VERBOSE means it will only display when verbose output is toggled.
+	 *
+	 * @param string|array $message  A string or a an array of strings to output
+	 * @param integer      $newlines Number of newlines to append
+	 * @param integer      $level    The message's output level, see above.
+	 *
+	 * @return integer|boolean Returns the number of bytes returned from writing to stdout.
+	 * @link http://book.cakephp.org/2.0/en/console-and-shells.html#Shell::out
+	 */
+	protected function _out($message = null, $tag = false, $newlines = 1, $level = Shell::NORMAL) {
+		if ($this->_debug === 1) {
+			if ($tag) {
+				$message = sprintf('<%1$s>%2$s</%1$s>', $tag, $message);
+			}
+			return $this->out($message, $newlines, $level);
+		}
+		# because we did not write anything to the stdout
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use "//" instead of "#" to start this comment [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://dev.expandonline.nl:9000/sonarqube/coding_rules#rule_key=php%3AS2046)